test/rest_api/test_storage_service.py

```diff
@@ -277,6 +277,14 @@ def verify_snapshot_details(expected):
                 'value': [{'ks': ks0, 'cf': cf00, 'total': 1, 'live': 0}]
             })
 
+            resp = rest_api.send("POST", f"storage_service/truncate/{ks0}", { 'cf': cf00 })
```

Comment:
Personally (but I guess this is just a matter of taste), I don't like tests to combine several completely unrelated features. It makes it harder to understand the test and what it really tests, and if in the future it fails, it makes it harder to understand which feature broke. I would have left the orginal test unmodified, and added a new test for a combination of snapshot and truncate - if you think there's something interesting in this combination.
But your call.

Reply:
Something I just thought about - this truncate/snapshot combination assumes that auto-snapshot is turned off? While test/cql-pytest/run indeed runs with auto-snapshot turns off, it was meant to be a performance improvement - not something that test correctness should rely on.

Reply:
I agree.

I have a follow up patch that adds an optional, boolean `snapshot` parameter to the truncate API
so the caller can override the default and explicitly say if a snapshot is desired or be skipped.

I should probably add a comment about the test assumption that auto-snapshot is off.

Reply:
> I agree.
> 
> I have a follow up patch that adds an optional, boolean `snapshot` parameter to the truncate API so the caller can override the default and explicitly say if a snapshot is desired or be skipped.

Since I think the truncate API shouldn't exist, I don't think this test should use it or improvements of it ;-)

> I should probably add a comment about the test assumption that auto-snapshot is off.

Even bette can be to *check* what is the setting of auto-snapshot, and set your expectations in the test accordingly. You can check the setting of a configuration parameter by reading from the `system.config` table - you can see an example in test/cql-pytest/conftest.py or test/cql-pytest/test_tools.py